x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/enrich/EnrichLookupOperator.java

```diff
@@ -107,9 +119,117 @@ protected void performAsync(Page inputPage, ActionListener<Page> listener) {
         );
     }
 
+    @Override
+    public String toString() {
+        return "EnrichOperator[index="
+            + enrichIndex
+            + " match_field="
+            + matchField
+            + " enrich_fields="
+            + enrichFields
+            + " inputChannel="
+            + inputChannel
+            + "]";
+    }
+
     @Override
     protected void doClose() {
         // TODO: Maybe create a sub-task as the parent task of all the lookup tasks
         // then cancel it when this operator terminates early (e.g., have enough result).
     }
+
+    @Override
+    public Operator.Status status() {
+        final PageStats stats = pageStats();
+        assert stats.received() >= stats.completed() : stats.received() + " < " + stats.completed();
+        return new Status(stats.received(), stats.completed(), totalTerms, TimeValue.timeValueNanos(totalTimeInNanos()).millis());
+    }
+
+    public static class Status implements Operator.Status {
+        public static final NamedWriteableRegistry.Entry ENTRY = new NamedWriteableRegistry.Entry(
+            Operator.Status.class,
+            "enrich",
+            Status::new
+        );
+
+        final long receivedPages;
+        final long completedPages;
+        final long totalTerms;
+        final long totalTimeInMillis;
+
+        Status(long receivedPages, long completedPages, long totalTerms, long totalTimeInMillis) {
+            this.receivedPages = receivedPages;
+            this.completedPages = completedPages;
+            this.totalTerms = totalTerms;
+            this.totalTimeInMillis = totalTimeInMillis;
+        }
+
+        Status(StreamInput in) throws IOException {
+            if (in.getTransportVersion().onOrAfter(TransportVersions.ESQL_ENRICH_OPERATOR_STATUS)) {
+                this.receivedPages = in.readVLong();
+                this.completedPages = in.readVLong();
+                this.totalTerms = in.readVLong();
+                this.totalTimeInMillis = in.readVLong();
+            } else {
+                this.receivedPages = -1L;
```

Comment:
I don't think it's possible to get here - if the other side doesn't know about the status it'll never send that named writeable.